framework/src/mesh/MooseMesh.C

```diff
@@ -1372,11 +1372,10 @@ MooseMesh::detectOrthogonalDimRanges(Real tol)
 
   // See if we matched all of the extremes for the mesh dimension
   this->comm().max(extreme_matches);
-  if (std::count(extreme_matches.begin(), extreme_matches.end(), true) != std::pow(2.0, (int)dim))
-    return false; // This is not a regular orthogonal mesh
+  if (std::count(extreme_matches.begin(), extreme_matches.end(), true) == std::pow(2.0, (int)dim))
```

Comment:
Hmm... I think this line is doing a potentially unsafe floating point equality comparison. If you look at [the docs for std::pow](https://en.cppreference.com/w/cpp/numeric/math/pow), version (7) says "If any argument has integral type, it is cast to double" and the return type is either double or long double. I'd change this to something like `Utility::pow<dim>(2)` which is guaranteed to return an integral type.

Reply:
Is dim known at compile time? If not just do 1 << dim

Reply:
Good point, `dim` is definitely a runtime variable.